### PR TITLE
[Conversation] workspaces and intents CRUD operations

### PR DESCRIPTION
**Pull request by boazdavid**
:new:
add support for new conversation api.
create/ update/ delete/ get/ list of: workspace/intent
includes javadoc, and unit tests.
code style was checked.

these functions of the api were not implemented:
example/ counterExamples

### CODE REVIEW

File: conversation/src/main/java/com/ibm/watson/developer_cloud/conversation/v1/model/workspace/nodes/CreateDialogNode.java [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+   */
+  public Object getMetadata() {
+    return metadata;
+  }
+
+  /**
+   * @param metadata
+   *          The metadata for the dialog node.
+   */
+  public void setMetadata(Object metadata) {
```
Comment (germanattanasio) [inline conversation/src/main/java/com/ibm/watson/developer_cloud/conversation/v1/model/workspace/nodes/CreateDialogNode.java]: Object (?) I think it should be `Map<String, Object>`

Comment (boazdavid) [inline conversation/src/main/java/com/ibm/watson/developer_cloud/conversation/v1/model/workspace/nodes/CreateDialogNode.java]: According to Conversation API it is an object. https://www.ibm.com/watson/developercloud/conversation/api/v1/?curl#create_workspace
```
metadata
object
Any metadata that is required by the workspace.
```
I did not find an example of that object.
Do you think we should represent all JSON objects as `Map<String, Object>`?

**Comment by germanattanasio** [comment]
@boazdavid, I'm looking at merging #646 instead of this. The code in that branch was autogenerated and it's aligned to the direction we are targeting with the SDK.
I will talk to @mkistler to make sure we have the tests and Javadoc you wrote here.
Sorry for the late notice.

**Comment by germanattanasio** [comment]
We are going to merge the code from @mkistler later today. I'm going to close this for now.

**Pull request closed without merging**